src/kademlia/dht_tracker.cpp

```diff
@@ -264,13 +244,12 @@ namespace libtorrent { namespace dht
 		// TODO: ideally this function would be called when the
 		// put completes
 		f();
-		return true;
 	}
 
-	bool put_mutable_item_callback(item& it, boost::function<void(item&)> cb)
+	void put_mutable_item_callback(item& it, boost::function<void(item&)> cb)
 	{
+		TORRENT_ASSERT(it.is_mutable());
 		cb(it);
-		return true;
```

Comment:
this means we want to perform a "put", right? Is there a reason to move that decision to have to be made up-front?